pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -226,52 +247,46 @@ func (g *Gatherer) updateCache(ctx context.Context) {
 
 func (g *Gatherer) updateAlertsCache(ctx context.Context, metricsClient rest.Interface) error {
 	const logPrefix = "conditional gatherer: "
+	klog.Info(logPrefix + "updating alerts cache for conditional gatherer")
 
 	g.firingAlerts = make(map[string][]AlertLabels)
 
-	data, err := metricsClient.Get().AbsPath("federate").
+	data, err := metricsClient.Get().
+		AbsPath("api/v1/query").
+		Param("query", "ALERTS").
 		Param("match[]", `ALERTS{alertstate="firing"}`).
 		DoRaw(ctx)
 	if err != nil {
 		return err
 	}
 
-	var parser expfmt.TextParser
-	metricFamilies, err := parser.TextToMetricFamilies(bytes.NewReader(data))
+	var response struct {
+		Data struct {
+			Results []struct {
+				Labels map[string]string `json:"metric"`
+			} `json:"result"`
+		} `json:"data"`
+	}
```

Comment:
Maybe time to use the Prometheus client types....

Reply:
I'll change it, thanks

Reply:
It's not super important for this PR. You can experiment in another one if you want. Not exactly sure, but perhaps https://github.com/prometheus/client_golang/blob/main/api/prometheus/v1/api.go